interfaces/apparmor/apparmor.go

```diff
@@ -83,22 +85,49 @@ func loadProfiles(fnames []string, cacheDir string, flags aaParserFlags) error {
 	return nil
 }
 
-// UnloadProfiles removes the named profiles from the running kernel.
-//
-// The operation is done with: apparmor_parser --remove $names...
-// The binary cache file is removed from /var/cache/apparmor
+// UnloadProfiles is meant to remove the named profiles from the running
+// kernel and then remove any cache files. Importantly, we can only unload
+// profiles when we are sure there are no lingering processes from the snap
+// (ie, forcibly stop all running processes from the snap). Otherwise, any
+// running processes will become unconfined. Since we don't have this guarantee
+// yet, leave the profiles loaded in the kernel but remove the cache files from
+// the system so the policy is gone on the next reboot. LP: #1818241
```

Comment:
I was not planning on tackling this. I'm of the opinion that we should not unload the profiles because a) it is the current behavior, b) that is what people are used to with non-snaps and c) I don't want to introduce complexity that might result in a snap command going unconfined. I understand that the snapd team is interested in removing this, so I don't object.

Reply:
I think we can explore doing this after the lifecycle tracking work gives us the tools to do this. You can count on me trying at least :)